### PR TITLE
Fix a truckload of clippy annotations, warnings and errors.

### PR DESCRIPTION
**Pull request by hirschenberger**
No offense, I know changing codestyle related things may offend devs. Just pick what seems to make sense to you.

Removing superfluous clones and references do make more sense.

But there are other things that seem to be serious, e.g. the unbound locks like this:

`let _ = Self::trigger_reindex(tables, reindex, self.path.as_path());`

### CODE REVIEW

File: src/db.rs [hunk 1]
```diff
@@ -1092,13 +1086,9 @@ impl IndexedChangeSet {
 			},
 		};
 		for (key, value) in self.changes.iter() {
-			match column.write_plan(key, value.as_ref().map(|v| v.as_slice()), writer)? {
-				// Reindex has triggered another reindex.
```
Comment (arkpar) [inline src/db.rs:1096]: This comment is lost

File: src/stats.rs [hunk 2]
```diff
@@ -88,12 +88,12 @@ impl ColumnStats {
 	pub fn from_slice(data: &[u8]) -> ColumnStats {
 		let mut cursor = Cursor::new(data);
 		let mut value_histogram: [AtomicU32; HISTOGRAM_BUCKETS] = unsafe { MaybeUninit::uninit().assume_init() };
-		for n in 0 .. HISTOGRAM_BUCKETS {
-			value_histogram[n] = read_u32(&mut cursor);
+		for item in value_histogram.iter_mut().take(HISTOGRAM_BUCKETS) {
```
Comment (arkpar) [inline src/stats.rs:91]: `take` is not need here it seems

**Comment by arkpar** [comment]
Thank you for the PR. Changes look good, apart from a couple of small nits. Formatting uses spaces instead of tabs for some of the new lines, but we are about to adopt cargofmt anyway, so that's not much of an issue.

Regarding `trigger_reindex`, dropping the lock won't cause any issues in both cases. Holding the lock is more robust however.

**Comment by hirschenberger** [comment]
I'll fix the nits and reformat.

**Pull request merged**